subprojects/core/src/integTest/groovy/org/gradle/api/tasks/CopyTaskIntegrationSpec.groovy

```diff
@@ -2239,4 +2240,187 @@ class CopyTaskIntegrationSpec extends AbstractIntegrationSpec {
         "rename(Transformer)"       | "rename(org.gradle.internal.Transformers.noOpTransformer())"
     }
 
+    def "renaming 2 different source files to the same name in the dest dir should give a clear error"() {
+        given: "a directory with a file"
+        def unzippedDir = file("before/files")
+        def unzippedFile = unzippedDir.file("sub/c.txt").touch()
+
+        and: "another directory with the same file"
+        def unzippedDir2 = file("before2/files")
+        def unzippedFile2 = unzippedDir2.file("sub/c2.txt").touch()
+
+        and: "a copy task that copies from both of these, failing on duplicates"
+        buildFile << """
+            task (copy, type: Copy) {
```

Comment:
🤔 why are you using this unusual way of declaring the task? Maybe you meant `task copy(type: Copy)`?

Reply:
Yep, just a typo.  I'm surprised that worked.